src/erlcloud_s3.erl

```diff
@@ -467,9 +468,17 @@ extract_acl(ACL) ->
     [extract_grant(Item) || Item <- ACL].
 
 extract_grant(Node) ->
-    [{grantee, extract_user(xmerl_xpath:string("Grantee", Node))},
+    [{grantee, extract_grantee(hd(xmerl_xpath:string("Grantee", Node)))},
      {permission, decode_permission(erlcloud_xml:get_text("Permission", Node))}].
 
+extract_grantee(Node) ->
```

Comment:
It's a bit confusing semantically: grantee is supposed to be extracted from xml node, not from a list of xml nodes containing exactly one element so it's up to a function caller to provide a valid argument here. extract_user has the interface you mentioned for no real reason